src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs

```diff
@@ -52,10 +52,9 @@ private unsafe byte[] EncryptOrDecrypt(byte[] data, RSAEncryptionPadding padding
 
             int modulusSizeInBytes = RsaPaddingProcessor.BytesRequiredForBitCount(KeySize);
 
-            if (!encrypt && data.Length > modulusSizeInBytes)
+            if (!encrypt && data.Length != modulusSizeInBytes)
             {
-                throw new CryptographicException(
-                    SR.Format(SR.Cryptography_Padding_DecDataTooBig, modulusSizeInBytes));
+                throw new CryptographicException(SR.Cryptography_RSA_DecryptWrongSize);
```

Comment:
Would it be helpful to include either or both data.Length / modulusSizeInBytes? (Same question for the other occurrences.)

Reply:
I thought about it.  It didn't feel terribly compelling, so I left it out.  Especially when decrypting you should have an idea of what your key is.  And you should only decrypt that which was encrypted.

That said, feedback later can lead to new conclusions :)